rfcs/convergence/authoring-stories.md

```diff
@@ -433,6 +436,46 @@ export const Default = (props: PopoverProps) => (
 Public stories should only contain code, which is useful for users to see after clicking on “Show code” in documentation.
 Extra markup (e.g., container with CSS styles) can be added via [Decorators](https://storybook.js.org/docs/react/writing-stories/decorators).
 
+### 10. Internal stories for testing
+
+As mentioned in for E2E testing, we should ensure maximum coverage for all publicly viewable stories by our consumers.
+For more complex scenarios that need to be tested we should make sure that stories exist for E2E tests but should not be
+easily accessible publicly.
+
+Storybook has proposed a feature for this in [storybookjs/storybook#9209](https://github.com/storybookjs/storybook/issues/9209)
+which will configure stories to exist in deeplink URL format, but do not appear in the nav tree or the docs page. As stated in the issue,
+we can workaround before the release of this feature by modifying `manager-head.html` and set `display:none`for all
+stories with a specific DOM `id` attribute. Storybook uses the `id` attribute for each link in the nav tree, and sets
+the value to the story id.
+
+We propose to use an extra filename extension and naming convention for internal stories:
+
+```ts
+// MenuTabstopsInternal.stories.tsx
+// Deep link /story/components-menu--tabstops-internal
+// Does not appear in the sidebar or docs page
+export const TabstopsInternal = () => {
+  // story
+};
+```
+
+The naming convention of the story simply adds the `Internal` keyword to the Pascal case story name. This will match the
+filename. More importantly the generated id will contain `menu-tabstops-internal`.
+
+We can simply use a css wildcard query selector:
+
+```css
+[id*='internal'] {
```

Comment:
Maybe this is just the engineer instinct to try to break everything 😆 but can this be made any more specific, like `[id$='-internal']`? ([reference](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors))

Reply:
yes of course, the idea was to just make the entire word `internal` a keyword, we can also constrain the position